app/models/detector/bulk_checker.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class Detector
+  # BulkTermChecker is expected to be added to Detectors via `extend Detector::BulkTermChecker` to allow the
+  # singleton class to access it
+  # See also: `PatternChecker` for shared instance methods
+  module BulkChecker
+    # This method is intended to be used for inspecting detections during development.
+    # Assumptions include
+    #   - the Class including this module implements a `detections` method (either via `attr_reader` or as a method)
+    #     that is only populated for Terms in which it has made a detection
+    #   - the initialize method accepts a `phrase` as a string
+    # @param output [boolean] optional. Defaults to false as that is the more likely scenario useful in development as
+    #   the logger output is often what is desired.
+    def check_all_matches(output: false)
+      count = 0
+      matches = []
+      Term.find_each do |t|
+        d = new(t.phrase)
+        next if d.detections.blank?
+
+        count += 1
+
+        matches.push [t.phrase, d.detections]
+      end
+
+      Rails.logger.ap matches
```

Comment:
You are correct it will throw an exception in production which is not great. I don't think it's worth loading the gem solely for this very-unlikely-to-be-used-in-prod feature, so I'm going to think a bit about whether it makes sense to just return a message stating it isn't available in production, or to have a path that doesn't use `ap` and instead just uses the build in object inspection outside of dev/test.